client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx

```diff
@@ -100,13 +125,84 @@ export function useGetProductVariants(
 		[ translate ]
 	);
 
-	return useMemo( () => productsWithPrices.map( getProductVariantFromAvailableVariant ), [
-		productsWithPrices,
-		getProductVariantFromAvailableVariant,
-	] );
+	const filteredVariants = useMemo( () => {
+		debug( 'found unfiltered variants', variantsWithPrices );
+		return variantsWithPrices.filter( ( product ) =>
+			isVariantAllowed( product, activePlan?.interval )
+		);
+	}, [ activePlan?.interval, variantsWithPrices ] );
+
+	const variantsWithComparativeDiscounts = useMemo(
+		() => addComparativeDiscountsToVariants( filteredVariants ),
+		[ filteredVariants ]
+	);
+
+	return useMemo( () => {
+		debug( 'found filtered variants', variantsWithComparativeDiscounts );
+		return variantsWithComparativeDiscounts.map( getProductVariantFromAvailableVariant );
+	}, [ getProductVariantFromAvailableVariant, variantsWithComparativeDiscounts ] );
+}
+
+function addComparativeDiscountsToVariants(
+	variants: AvailableProductVariant[]
+): AvailableProductVariantAndCompared[] {
+	return variants.map( ( variant ) => {
+		return {
+			...variant,
+			priceFullBeforeDiscount: getLowestPriceTimesVariantInterval( variant, variants ),
+		};
+	} );
 }
 
-function VariantPrice( { variant }: { variant: AvailableProductVariant } ) {
+function getLowestPriceTimesVariantInterval(
+	variant: AvailableProductVariant,
+	allVariants: AvailableProductVariant[]
+): number {
+	if ( allVariants.length < 1 ) {
+		throw new Error(
+			'There must be at least one variant to compare against when generating relative prices'
+		);
+	}
+	allVariants.sort( ( variantA, variantB ) => {
+		const variantAInterval = getTermDuration( variantA.plan.term );
+		const variantBInterval = getTermDuration( variantB.plan.term );
+		return variantAInterval - variantBInterval;
+	} );
+	const lowestVariant = allVariants[ 0 ];
+	const lowestVariantInterval = getTermDuration( lowestVariant.plan.term );
+	const variantInterval = getTermDuration( variant.plan.term );
+	const lowestVariantIntervalsInVariantInterval = Math.ceil(
+		variantInterval / lowestVariantInterval
```

Comment:
This works, but partially by coincidence... it basically works because we're representing a month term as 31 days right now.  If it were 30 days instead (a reasonable representation), then you'd have `Math.ceil( 365 / 30 ) = 13`, which is wrong.

I think it might be better to use `getBillingMonthsForTerm()` to figure out the exact ratio, similar to what the previous code did.